src/provider/index.ts

```diff
@@ -134,6 +139,20 @@ export class Main {
                     };
                     this._eventPump.push<NotificationActionEvent>(source.uuid, event);
                 }
+            } else if (action instanceof ExpireNotification) {
+                const {notification, source} = action.notification;
+
+                if (notification.onExpire !== null) {
+                    const event: Targeted<Transport<NotificationActionEvent>> = {
+                        target: 'default',
+                        type: 'notification-action',
+                        trigger: ActionTrigger.EXPIRE,
+                        notification: mutable(notification),
+                        result: notification.onExpire
+                    };
+                    this._eventPump.push<NotificationActionEvent>(source.uuid, event);
+                }
+                this._store.dispatch(new RemoveNotifications([action.notification]));
```

Comment:
Muddle if you're talking about actions in the Store sense, or actions in the `notification-action` sense. Talk when you're in

Reply:
This was "action" in the "store action" sense. Resolving based on earlier discussion.

To summarise discussion: there's no concept of "close reason" in the API (and we can't use `TriggerAction` for this, without violating the constraint of "all notifications will at some point get a `TriggerAction.CLOSE` action), so a sepatate "expiry" action is the only way for applications to have any idea that a notification has expired - should that be something they're interested in.

We'll need to consider the implications of this when we introduce programmatic actions - to decide how to handle the case of an 'expiry' action attempting to prevent the notification closing, as a button click action might do. But ultimately, this would be another instance of a problem that exists already, rather than introducing a new problem.